mblack edits (#173)
ni/install-systemlink-enterprise: release-notes/2023-11/README.md

# SystemLink Enterprise release 2023-11 Release Notes

> The 2023-11 release for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.

Reviewer: Probably want to update the release notes template as well: https://github.com/ni/install-systemlink-enterprise/blob/main/chapter-template/release-notes-template.md

Author: Good call. I'll include that change here since it's so small.

## New Features and Behavior changes

- Work Orders
    - The Work Order's expose a new privilege for scheduling test plans.
    - The Work Order service and UI is not included in the SystemLink Helm chart and associated privileges can be ignored.

## Helm Chart Breaking Changes

- `sl-jupyterhub`
    - Notebooks now support Python 3.11 in the JupyterHub development environment.

- `nbexecservice`
    - Notebooks now support Python 3.11 in notebooks executed by Routines.

- `testmonitorservice 0.16.37`

>     - The update includes a new database schema migration to support future features. This migration adds a new column to the `results` table. The migration is expected to complete without downtime. Downgrading to prior versions of the TestMonitor service after updating to this version is not supported.

Reviewer 2: I include "This migration adds a new column to the `results` table." in the excel sheet, but I don't think customers really need to know that. In our other documentation we say we will change the schema without notice on what we are changing. Up to you whether we leave it or not.

Author: Ok, I was on the fence about including this. I will drop. Thanks for the confirmation.

>     - The migration requires permission for `ALTER TABLE`, which is an elevated privileges beyond the minimal set required by the service's normal operation. To use `ALTER TABLE` the PostgreSQL user must be the owner of the table or be a member of the role that owns the table. If your PostgreSQL user does not have the required permissions the updated pods job will fail to start. See below for new options to facilitate this requirement.

Reviewer 2: Similar to my comment above, this might be a step beyond the level of detail we need.
```suggestion
    - The migration requires permission to alter tables in the Test Monitor schema, which is an elevated privileges beyond the minimal set required by the service's normal operation. To alter tables the PostgreSQL user must be the owner of the schema or be a member of the role that owns the schema. If your PostgreSQL user does not have the required permissions the updated pods job will fail to start. See below for new options to facilitate this requirement.
```

Author: Will use

### Setting up an Elevated PostgreSQL User for Migration

- The database migrations is performed by a Kubernetes job that runs prior to TestMonitor pod deployment.

> - The `testmonitorservice` Helm chart includes a new options to specify the PostgreSQL user for migration separately from the user for the service's normal operation. This allows the higher privileged user to be used during application updates exclusively.

Reviewer: "includes a new options" > "includes new options" or "includes a new option"

> - The `testmonitorservice` Helm chart includes a new options to specify the PostgreSQL user for migration separately from the user for the service's normal operation. This allows the higher privileged user to be used during application updates exclusively.

Reviewer: We might want to call out that setting a separate user for migration is optional.

Reviewer 2: I agree, these read like these new configuration options are required, which they are not. If you only use the prior configuration that will apply to both to the service replicas and the migration.

I also have a PR to update the ni.com docs with detailed instructions on how to configure these user credentials. That needs review from @MelissaHilliard . If that update is approved we should link to that document here. It will be an update to [this document](https://www.ni.com/docs/bundle/systemlink-enterprise/page/config-systemlink-enterprise.html#GUID-22D7F822-FF82-436A-9458-DA1D33334886__GUID-A2CA66A5-7E59-419B-8638-48E7E7A3963A).

[Pull Request 590739](https://dev.azure.com/ni/Users/_git/Documentation%20Reviews/pullrequest/590739): Expand docs on configuring postgres users for TestMonitor

Author: @MelissaHilliard do you think this document would be published this week so we could refer to it in the release notes? Do you think we should lift the current wording and structure that will be used in the ni.com help into the release notes?

- New secrets must be defined in Helm for this PostgreSQL user
    - `testmonitorservice.secrets.database.migrationConnectionString` and `testmonitorservice.secrets.database.migrationConnectionPassword`.
    - [View this configuration](https://github.com/ni/install-systemlink-enterprise/blob/2023-11/getting-started/templates/systemlink-secrets.yaml#L434)

> - These secretes and a new connection string must be referenced in your values file.

Reviewer: "secretes" > "secrets"

The 2023-11 release for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.

## New Features and Behavior changes

- Work Orders
    - The Work Order's expose a new privilege for scheduling test plans.
    - The Work Order service and UI is not included in the SystemLink Helm chart and associated privileges can be ignored.

## Helm Chart Breaking Changes

- `sl-jupyterhub`
    - Notebooks now support Python 3.11 in the JupyterHub development environment.

- `nbexecservice`
    - Notebooks now support Python 3.11 in notebooks executed by Routines.

- `testmonitorservice 0.16.37`
    - The update includes a new database schema migration to support future features. The migration is expected to complete without downtime. Downgrading to prior versions of the TestMonitor service after updating to this version is not supported.

>     - The migration requires permission to alter tables in the Test Monitor schema, which is an elevated privileges beyond the minimal set required by the service's normal operation. To alter tables the PostgreSQL user must be the owner of the schema or be a member of the role that owns the schema. If your PostgreSQL user does not have the required permissions the updated pods job will fail to start. See below for new options to facilitate this requirement.

Reviewer 2: ```suggestion
    - The migration requires permission to alter tables in the Test Monitor schema, which are elevated privileges beyond the minimal set required by the service's normal operation. To alter tables the PostgreSQL user must be the owner of the schema or be a member of the role that owns the schema. If your PostgreSQL user does not have the required permissions the updated pods job will fail to start. See below for new options to facilitate this requirement.
```
